Jellyfin/Core/FullScreenManager.cs

```diff
@@ -0,0 +1,197 @@
+        private static Predicate<HdmiDisplayMode> ToHdmiDisplayHdrOptionMatchesPredicate(
+            HdmiDisplayHdrOption hdmiDisplayHdrOption)
+        {
+            bool HdrMatches(HdmiDisplayMode mode) => hdmiDisplayHdrOption == HdmiDisplayHdrOption.None ||
+                                                             (hdmiDisplayHdrOption ==
+                                                              HdmiDisplayHdrOption.DolbyVisionLowLatency &&
+                                                              mode.IsDolbyVisionLowLatencySupported) ||
+                                                             (hdmiDisplayHdrOption == HdmiDisplayHdrOption.Eotf2084 &&
+                                                              mode.IsSmpte2084Supported);
+            return HdrMatches;
+        }
+
+        private HdmiDisplayMode GetBestDisplayMode(uint videoWidth, uint videoHeight, double videoFrameRate, HdmiDisplayHdrOption hdmiDisplayHdrOption)
+        {
+            HdmiDisplayInformation hdmiDisplayInformation = HdmiDisplayInformation.GetForCurrentView();
+            IEnumerable<HdmiDisplayMode> supportedHdmiDisplayModes = hdmiDisplayInformation.GetSupportedDisplayModes();
+            // `GetHdmiDisplayHdrOption(...)` ensures the HdmiDisplayHdrOption is always a mode the display supports
+            Predicate<HdmiDisplayMode> hdrMatchesPredicate =
+                ToHdmiDisplayHdrOptionMatchesPredicate(hdmiDisplayHdrOption);
+            IEnumerable<HdmiDisplayMode> hdmiDisplayModes =
+                supportedHdmiDisplayModes.Where(mode => hdrMatchesPredicate.Invoke(mode));
+
+            bool filteredToVideoResolution = false;
+            if (Central.Settings.AutoResolution)
+            {
+                Predicate<HdmiDisplayMode> resolutionMatchesVideoPredicate =
+                    ToResolutionMatchesPredicate(videoWidth, videoHeight);
+                IEnumerable<HdmiDisplayMode> matchingResolution =
+                    hdmiDisplayModes.Where(mode => resolutionMatchesVideoPredicate.Invoke(mode));
+                if (matchingResolution.Any())
+                {
+                    hdmiDisplayModes = matchingResolution;
+                    filteredToVideoResolution = true;
+                }
+            }
+
+            HdmiDisplayMode currentHdmiDisplayMode = hdmiDisplayInformation.GetCurrentDisplayMode();
+            Predicate<HdmiDisplayMode> resolutionMatchesCurrentDisplayPredicate = ToResolutionMatchesPredicate(
+                currentHdmiDisplayMode.ResolutionWidthInRawPixels,
+                currentHdmiDisplayMode.ResolutionHeightInRawPixels);
+            if (!filteredToVideoResolution)
+            {
+                hdmiDisplayModes =
+                    hdmiDisplayModes.Where(mode => resolutionMatchesCurrentDisplayPredicate.Invoke(mode));
+            }
+
+            if (Central.Settings.AutoRefreshRate)
+            {
+                Predicate<HdmiDisplayMode> refreshRateMatchesVideoPredicate =
```

Comment:
it does not really make sense to have that as a delegate here. Just skip the delegate and invoke the method directly